src/core/src/validation_util.cpp

```diff
@@ -1188,6 +1188,15 @@ std::shared_ptr<ov::op::v0::Constant> ov::util::constantfold_subgraph(const Outp
     if (num_inputs == 0)
         return nullptr;
 
+    if (subgraph_sink.get_tensor().has_and_set_bound()) {
+        const auto& lower = subgraph_sink.get_tensor().get_lower_value();
```

Comment:
Does this impact models with nested sub-graphs like inside `If` and `Loop` only or does it have wider impact?

Reply:
It impacts any model with MultiSubGraphOp.